src/serde.rs

```diff
@@ -0,0 +1,94 @@
+extern crate serde;
+
+use std::{cmp, fmt};
+use std::marker::PhantomData;
+use self::serde::{Serialize, Serializer, Deserialize, Deserializer, de};
+use super::{Bytes, BytesMut};
+
+macro_rules! serde_impl {
+    ($ty:ident, $visitor_ty:ident) => (
+        impl Serialize for $ty {
+            #[inline]
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                where S: Serializer
+            {
+                serializer.serialize_bytes(&self)
+            }
+        }
+
+        struct $visitor_ty<$ty> {
+            marker: PhantomData<$ty>,
+        }
+
+        impl<'de> de::Visitor<'de> for $visitor_ty<$ty> {
+            type Value = $ty;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("byte array")
+            }
+
+            #[inline]
+            fn visit_unit<E>(self) -> Result<Self::Value, E>
```

Comment:
Please remove this method. Questionable implicit conversions like this were removed in Serde 1.0.